addons/utm/models/utm_source.py

```diff
@@ -72,12 +72,17 @@ def create(self, vals_list):
         return super().create(vals_list)
 
     def write(self, values):
-        if values.get(self._rec_name) and not values.get('name'):
-            values['name'] = self.env['utm.source']._generate_name(self, values[self._rec_name])
-        if values.get('name'):
-            values['name'] = self.env['utm.mixin']._get_unique_names(self._name, [values['name']])[0]
-
-        super().write(values)
+        self.ensure_one()
```

Comment:
@reth-odoo ah right. My logic here was: if i have multiple mailings and i am trying to write same name for all of them, then it will cause an error because utm.source have unique name constraint.
But now i remember that that constraint is about to get merged in master and in other stable versions we add [n] at the end of the duplicate source, so this one is valid change on all stable versions but not for **master**

Reply:
@reth-odoo but i think it will still cause an error cuz write function isn't overriden. so i think it's good idea to do ensure_one